devito/operator/operator.py

```diff
@@ -894,14 +894,20 @@ def apply(self, **kwargs):
         >>> op = Operator(Eq(u3.forward, u3 + 1))
         >>> summary = op.apply(time_M=10)
         """
+        # Compile the operator before building the arguments list
+        # to avoid out of memory with greedy compilers
+        try:
+            cfunction = self.cfunction
+        except Exception as e:
```

Comment:
drop try-except; `compiler.py`'s already throws an exception if compilation fails, and that's what the user should see

Reply:
done